{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyN4OUCkx8GCJnwv1uj4qVkI",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/gordesakshi202-ui/Dna_sequence_analysis/blob/main/Copy_of_Dna_Sequence__Analysis.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sb3m88kXTd8u",
        "outputId": "bc0ac9f7-9c69-4683-85b2-6953d517be2d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Sequence Length: 48\n",
            "A: 6\n",
            "T: 6\n",
            "G: 18\n",
            "C: 18\n",
            "GC Content (%): 75.0\n"
          ]
        }
      ],
      "source": [
        "\n",
        "#initialize empty string to store DNA sequence\n",
        "sequence = \"\"\n",
        "\n",
        "#read FASTA file and ignore header lines\n",
        "with open(\"/content/brca1.fasta\", \"r\") as file:\n",
        "    for line in file:\n",
        "        if not line.startswith(\">\"):\n",
        "            sequence += line.strip().upper()\n",
        "\n",
        "#calculate sequence length\n",
        "length = len(sequence)\n",
        "\n",
        "#count the number of nucleotides\n",
        "A = sequence.count(\"A\")\n",
        "T = sequence.count(\"T\")\n",
        "G = sequence.count(\"G\")\n",
        "C = sequence.count(\"C\")\n",
        "\n",
        "#calculate the gc content\n",
        "gc_content = ((G + C) / length) * 100\n",
        "\n",
        "#print results\n",
        "print(\"Sequence Length:\", length)\n",
        "print(\"A:\", A)\n",
        "print(\"T:\", T)\n",
        "print(\"G:\", G)\n",
        "print(\"C:\", C)\n",
        "print(\"GC Content (%):\", gc_content)\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## DNA Sequence Analysis using NCBI Data and Python\n",
        "## Aim\n",
        "to analyze the dna sequence using python and calculate GC content.\n",
        "### Methods\n",
        "\n",
        "1. The DNA sequence of the BRCA1 gene was retrieved from the NCBI nucleotide database in FASTA format.  \n",
        "2. The FASTA file was uploaded and read using Python in Google Colab.  \n",
        "3. Header lines starting with \">\" were ignored, and the nucleotide sequence was extracted.  \n",
        "4. Python was used to calculate the total sequence length and count the number of adenine (A), thymine (T), guanine (G), and cytosine (C) bases.  \n",
        "5. GC content was calculated using the formula: (G + C) / total length Ã— 100.  \n",
        "6. The results were printed and interpreted to understand nucleotide composition.\n",
        "###### Results\n",
        "\n",
        "The analyzed DNA sequence had a length of 48 nucleotides.  \n",
        "The nucleotide composition showed higher GC content compared to AT content.  \n",
        "The GC content of the sequence was calculated to be 75%.\n",
        "## Conclusion\n",
        "\n",
        "This analysis demonstrates how Python can be used for basic bioinformatics tasks such as DNA sequence analysis.  \n",
        "GC content calculation helps in understanding the stability and characteristics of the DNA sequence."
      ],
      "metadata": {
        "id": "TNtsoe_FY_az"
      }
    }
  ]
}